src/utils/searchHelpers.ts

```diff
@@ -0,0 +1,32 @@
+import { RegistrationSubtype } from '../types/publicationFieldNames';
+
+interface PropertySearch {
+  key: string;
+  value: string;
+}
+interface SearchConfig {
+  searchTerm?: string;
+  subtypes?: RegistrationSubtype[];
+  properties?: PropertySearch[];
+}
+
+const createSearchTermFilter = (searchTerm?: string) => (searchTerm ? `*${searchTerm}*` : '');
+
+const createSubtypeFilter = (subtypes?: RegistrationSubtype[]) =>
+  subtypes && subtypes.length > 0
+    ? `(${subtypes.map((subtype) => `entityDescription.reference.publicationInstance="${subtype}"`).join(' OR ')})`
+    : '';
+
+const createPropertyFilter = (properties?: PropertySearch[]) =>
+  properties && properties.length > 0
+    ? `(${properties.map(({ key, value }) => `${key}="${value}"`).join(' OR ')})`
+    : '';
+
+export const createSearchQuery = (searchConfig: SearchConfig) => {
+  const textSearch = createSearchTermFilter(searchConfig.searchTerm);
+  const typeSearch = createSubtypeFilter(searchConfig.subtypes);
+  const propertySearch = createPropertyFilter(searchConfig.properties);
+
+  const searchQuery = [textSearch, typeSearch, propertySearch].filter((search) => !!search).join(' AND ');
+  return searchQuery;
+};
```

Comment:
kanskje er det litt rart å separere typeSearch og propertySearch, siden man kan bruke propertySearch for å søke på typer også... fjerne typeSearch?
```
createSearchQuery({
   properties: [
      { 
          key: "entityDescription.reference.publicationInstance",
          value: "JournalArticle"
       },
      { 
          key: "entityDescription.reference.publicationInstance",
          value: "ShortCommunication"
       }
   ]
})
```
er det samme som
```
createSearchQuery({
   subtypes: [ "JournalArticle" , "ShortCommunication]
})
```

Reply:
kanskje kan vi være så lur å tillate at `value` kan være array når man søker for properties også 🤔

Reply:
fant ut nå at `query=(entityDescription.reference.publicationInstance=("JournalArticle" AND "JournalShortCommunication"))` er det samme som `query=(entityDescription.reference.publicationInstance="JournalArticle" OR entityDescription.reference.publicationInstance="JournalShortCommunication")` så da gir det i hvert fall mening å fjerne typeSearch